### PR TITLE
Support Java 17+ compiled Beam components for build, release, and xlang tests

### PR DESCRIPTION
**Pull request by Abacn**
On Beam SDK side:

Honor "JAVA_HOME" environment variable in multiple places. This actually does not change non-test behavior, as JavaServer already honors "JAVA_HOME":

https://github.com/apache/beam/blob/e24911c5b75dff57817a8b602645b7bc4348bb92/sdks/python/apache_beam/utils/subprocess_server.py#L280

Currently if JAVA_HOME is different from PATH java, there is an inconsistency such that certain test spins up a (Flink/Spark) job server with default Java, but expansion service runs on JAVA_HOME java

On test infra side:

Support Beam components requiring Java17+ for release workflows They will be compiled with JDK21 with byte code compatibility configured by applyJavaNature(requireJavaVersion)

Python xlang tests now also support fork java versions, like we did for Java builds in #34858. Techinically run with single, latest LTS Java (Java21) should also work, but not tested.

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
convert to draft for now. This should get merged after 2.66 release candidate built, to give it time exercise PostSubmit tests

File: sdks/python/apache_beam/yaml/yaml_provider.py [hunk 1]
```diff
@@ -385,7 +385,8 @@ def __init__(
 
   def available(self):
     # pylint: disable=subprocess-run-check
-    trial = subprocess.run(['which', 'java'], capture_output=True)
+    trial = subprocess.run(['which', subprocess_server.JavaHelper.get_java()],
```
Comment (Abacn) [inline sdks/python/apache_beam/yaml/yaml_provider.py:388]: there was an edge case here: if only JAVA_HOME is provided, the external transform expansion should work, but here it assumes java not available

**Comment by Abacn** [comment]
Test looks good: https://github.com/apache/beam/actions/runs/15614582305/job/43983947372 (update: other than 503s due to global GCP outage...)

File: sdks/python/apache_beam/io/external/xlang_debeziumio_it_test.py [hunk 2]
```diff
@@ -34,12 +36,32 @@
...
 
+def _disable_debezium_test():
+  # disable if run on <Java17
+  try:
+    java = subprocess_server.JavaHelper.get_java()
+    result = subprocess.run([java, '-version'],
+                            check=True,
+                            capture_output=True,
+                            text=True)
+    version_line = result.stderr.splitlines()[0]
```
Comment (damccorm) [inline sdks/python/apache_beam/io/external/xlang_debeziumio_it_test.py:47]: It would probably help to have a comment with the expected output here (`openjdk version "21.0.6" 2025-01-21`)

Comment (Abacn) [reply sdks/python/apache_beam/io/external/xlang_debeziumio_it_test.py:47]: Thanks, added

**Comment by Abacn** [comment]
Let me merge this to be able to get DebeziumIO change in. The .github change should not affect ongoing release process
